src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -1123,7 +1128,7 @@ private Symbol UseNonFieldSymbolUnsafely(BoundExpression expression)
                     case BoundKind.FieldAccess:
                         {
                             var fieldAccess = (BoundFieldAccess)expression;
-                            var fieldSymbol = fieldAccess.FieldSymbol;
+                            var fieldSymbol = fieldAccess.FieldSymbol.OriginalDefinition;
```

Comment:
>var fieldSymbol = fieldAccess.FieldSymbol.OriginalDefinition; [](start = 28, length = 61)

I think it would be better to adjust implementation of NoteFieldAccess